src/Sarif.Multitool/Rules/SARIF2012.ProvideHelpUris.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+using Microsoft.Json.Pointer;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class ProvideHelpUris : SarifValidationSkimmerBase
+    {
+        /// <summary>
+        /// SARIF2012
+        /// </summary>
+        public override string Id => RuleId.ProvideHelpUris;
+
+        /// <summary>
+        /// Placeholder
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF2012_ProvideHelpUris_FullDescription_Text };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+            nameof(RuleResources.SARIF2012_ProvideHelpUris_Note_Default_Text)
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Note;
+
+        protected override void Analyze(Tool tool, string toolPointer)
```

Comment:
>Analyze [](start = 32, length = 7)

Let's back up one more level, to the `run` level. There are two `Tool` objects in SARIF, and one of them is related to conversion from a tool's native output format to SARIF. We're not interested in what the converter has to say.

Reply:
We're only interested in `run.tool.driver.rules`.

---
In reply to: [447985445](https://github.com/microsoft/sarif-sdk/pull/1949#discussion_r447985445) [](ancestors = 447985445)

Reply:
This is fine. By the way, @hakohli@microsoft.com  has a different way of following these nestings. See her most recent PR. Just FYI.

---
In reply to: [447987255](https://github.com/microsoft/sarif-sdk/pull/1949#discussion_r447987255) [](ancestors = 447987255,447985445)